pkg/apis/core/validation/utils.go

```diff
@@ -195,3 +195,52 @@ func ValidateIPFamilies(ipFamilies []core.IPFamily, fldPath *field.Path) field.E
 
 	return allErrs
 }
+
+func validateDependencyWatchdogSettings(settings *core.SeedSettingDependencyWatchdog, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if settings.Endpoint != nil && settings.Weeder != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "only one of spec.settings.dependencyWatchdog.weeder and spec.settings.dependencyWatchdog.endpoint can be specified, not both."))
```

Comment:
The error message already contains fldPath, We don't need to put the whole spec path.
```suggestion
		allErrs = append(allErrs, field.Forbidden(fldPath, "only one of weeder and endpoint can be specified, not both."))
```